### DISCUSSION TITLE
Capture proxied auth header and use in subsequent proxied requests?

### DISCUSSION THREAD

**Discussion by ghost** [mockoon/mockoon#1104] [Help] [answered]
I'm using partial proxying so that my login endpoint is never proxied (or at least sometimes proxied) so that I can get a valid auth token from the server. This is necessary so that any unmocked endpoints that get proxied use the correct auth token header. The problem is, after I login via the proxy, any already mocked endpoints with previously recorded auth header are overwriting the new token with the previously recorded mock token.

Is there a way to capture the auth header from my initial proxied login and then use templating to be sure that header gets used for any proxied requests?

**Comment by 255kb** [comment, marked as answer]
This is not yet possible. We would need some way to reference a response's body or headers from the templating system (which is supported in headers names and values). 
I don't think there is a opened issue for this.

**Reply by ghost** [reply]
I'd like to explore submitting a PR for this feature. I'll open an issue and dig in a bit to see how feasible this is. Thanks for the reply.

**Reply by 255kb** [reply]
Sure, feel free to reach me on Discord (my username is 255kb) if you want to get a tour of the code. I also have an idea about how to implement this without too much hassle. Not saying it's completely hassle-free though!
